fs/fs-fuzz.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM
+# Author:Praveen K Pandey <praveen@linux.vnet.ibm.com>
+#
+
+
+import os
+
+from avocado import Test
+from avocado import main
+from avocado.utils import process, archive, build
+from avocado.utils.software_manager import SoftwareManager
+
+
+class FsFuzz(Test):
+
+    """
+    fs-fuzz : Two simple fuzzers, one for  filesystem operations
+    """
+
+    def clear_dmesg(self):
+        process.run("dmesg -c ", sudo=True)
+
+    def verify_dmesg(self):
+        self.whiteboard = process.system_output("dmesg")
+        pattern = ['WARNING: CPU:', 'Oops',
+                   'Segfault', 'soft lockup', 'Unable to handle']
+        for fail_pattern in pattern:
+            if fail_pattern in self.whiteboard:
+                self.fail("Test Failed : %s in dmesg" % fail_pattern)
+
+    def setUp(self):
+        '''
+        Build FileBench
+        Source:
+        https://github.com/regehr/fs-fuzz
+        '''
+
+        # Check for basic utilities
+        smm = SoftwareManager()
+
+        for package in ['make', 'gcc']:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+        tarball = self.fetch_asset(
+            'https://github.com/regehr/fs-fuzz/archive/master.zip')
+
+        archive.extract(tarball, self.workdir)
+        self.build_dir = os.path.join(self.workdir, 'fs-fuzz-master')
+        build.make(self.build_dir)
+        self.clear_dmesg()
+
+    def test(self):
+        os.chdir(self.build_dir)
+
+        if process.system('./fd_fuzz 1', shell=True, ignore_status=True) != 0:
+            self.fail("stress test for  file I/O layer failed")
+        if process.system('./file_fuzz 1', shell=True, ignore_status=True) != 0:
```

Comment:
I would suggest adding 2 separate `test_x()` for this.

Reply:
@narasimhan-v ah yes cool suggestion will adopt